core/shared/src/main/scala/fs2/Compiler.scala

```diff
@@ -34,31 +34,29 @@ import scala.annotation.implicitNotFound
   "Cannot find an implicit Compiler[F, G]. This typically means you need a Concurrent[F] in scope"
 )
 sealed trait Compiler[F[_], G[_]] {
-  private[fs2] def apply[O, B, C](stream: Pull[F, O, Unit], init: () => B)(
-      fold: (B, Chunk[O]) => B,
-      finalize: B => C
-  ): G[C]
+  private[fs2] val target: Monad[G]
+  private[fs2] def apply[O, B](stream: Pull[F, O, Unit], init: B)(
+      fold: (B, Chunk[O]) => B
+  ): G[B]
 }
 
 private[fs2] trait CompilerLowPriority2 {
 
   implicit def resource[F[_]: Compiler.Target]: Compiler[F, Resource[F, *]] =
     new Compiler[F, Resource[F, *]] {
-      def apply[O, B, C](
+      val target: Monad[Resource[F, *]] = implicitly
+      def apply[O, B](
           stream: Pull[F, O, Unit],
-          init: () => B
-      )(foldChunk: (B, Chunk[O]) => B, finalize: B => C): Resource[F, C] =
+          init: B
+      )(foldChunk: (B, Chunk[O]) => B): Resource[F, B] =
         Resource
           .makeCase(CompileScope.newRoot[F])((scope, ec) => scope.close(ec).rethrow)
           .flatMap { scope =>
             def resourceEval[A](fa: F[A]): Resource[F, A] =
```

Comment:
Since you're at it, I've fixed this in ce some time ago, we can delete it

Reply:
Cool, fixed with an evalMap